File: LeetCode75-sols/Array_String/JavaScript/Greatest_No_Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    const maxCandies = Math.max(...candies);
    const result = candies.map(candy => {
        return (candy + extraCandies >= maxCandies);
    });

    return result;
};

File: LeetCode75-sols/Array_String/C#/Greatest_No_Candies.cs
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        // Find max no of candies a kid has
        int maxCandies = candies.Max();

        List<bool> result = new List<bool>();

        foreach (int candy in candies) {
            if (candy + extraCandies >= maxCandies) {
                result.Add(true);
            } else {
                result.Add(false);
            }
        }
        return result;
    }
}
